### ISSUE TITLE
Spark3: Unparsable section when `WHERE` predicate literal is wrapped in double-quotes

### ISSUE THREAD

**Issue by R7L208** [sqlfluff/sqlfluff#2309]
The below code produces an `unparsable section` in spark3 dialect.

```sql
SELECT * FROM student WHERE name = "John Smith";
```

However, if double-quotes are changed to single-quotes, it parses correct in spark3.

```sql
SELECT * FROM student WHERE name = 'John Smith';
```

Similarly if `sqlfluff lint` is run with `ansi` dialect, it produces correct output with either double- or single-quotes.

```bash
> sqlfluff lint --dialect ansi test/fixtures/dialects/spark3/temp.sql
== [test/fixtures/dialects/spark3/temp.sql] FAIL
L:   1 | P:   1 | L044 | Query produces an unknown number of result columns.
L:   1 | P:  36 | L057 | Do not use special characters in identifiers.
All Finished 📜 🎉!
```

## Expected Behaviour

```bash
> sqlfluff lint --dialect spark3 test/fixtures/dialects/spark3/temp.sql
== [test/fixtures/dialects/spark3/temp.sql] FAIL
L:   1 | P:   1 | L044 | Query produces an unknown number of result columns.
L:   1 | P:  36 | L057 | Do not use special characters in identifiers.
All Finished 📜 🎉!
```

## Observed Behaviour

```bash
==== sqlfluff ====
sqlfluff:                0.9.1 python:                 3.10.1
implementation:        cpython dialect:                spark3
verbosity:                   3 templater:               jinja

== Raw Config:
core:                                   
    dialect:            spark3              
    disable_noqa:       False               
    encoding:           autodetect          
    ignore:             []                  
    ignore_templated_areas:True                
    nocolor:            False               
    output_line_length: 80                  
    recurse:            True                
    runaway_limit:      10                  
    sql_file_exts:      .sql,.sql.j2,.dml,.ddl
    templater:          jinja               
    verbose:            3                   
indentation:                            
    indented_joins:     False               
    indented_using_on:  True                
    template_blocks_indent:True                
rules:                                  
    allow_scalar:       True                
    comma_style:        trailing            
    indent_unit:        space               
    max_line_length:    80                  
    single_table_references:consistent          
    tab_space_size:     4                   
    unquoted_identifiers_policy:all                 
    Example_L001:                           
        forbidden_columns:  bar, baaz           
    L007:                                   
        operator_new_lines: after               
    L010:                                   
        capitalisation_policy:consistent          
    L011:                                   
        aliasing:           explicit            
    L012:                                   
        aliasing:           explicit            
    L014:                                   
        extended_capitalisation_policy:consistent          
    L016:                                   
        ignore_comment_lines:False               
    L026:                                   
        force_enable:       False               
    L028:                                   
        force_enable:       False               
    L029:                                   
        unquoted_identifiers_policy:aliases             
    L030:                                   
        capitalisation_policy:consistent          
    L038:                                   
        select_clause_trailing_comma:forbid              
    L040:                                   
        capitalisation_policy:consistent          
    L042:                                   
        forbid_subquery_in: join                
    L047:                                   
        prefer_count_0:     False               
        prefer_count_1:     False               
    L052:                                   
        multiline_newline:  False               
        require_final_semicolon:False               
    L054:                                   
        group_by_and_order_by_style:consistent          
    L057:                                   
        additional_allowed_characters:""                  
        allow_space_in_identifier:False               
        quoted_identifiers_policy:all                 
        unquoted_identifiers_policy:all                 
templater:                              
    unwrap_wrapped_queries:True                
    jinja:                                  
        apply_dbt_builtins: True                
        macros:                                 
            dbt_config:         {% macro config() %}{% for k in kwargs %}{% endfor %}{% endmacro %}
            dbt_is_incremental: {% macro is_incremental() %}True{% endmacro %}
            dbt_ref:            {% macro ref(model_ref) %}{{model_ref}}{% endmacro %}
            dbt_source:         {% macro source(source_name, table) %}{{source_name}}_{{table}}{% endmacro %}
            dbt_var:            {% macro var(variable, default='') %}item{% endmacro %}

==== readout ====

=== [ path: test/fixtures/dialects/spark3/temp.sql ] ===

INFO       TEMPLATING RAW [jinja] (test/fixtures/dialects/spark3/temp.sql)
INFO       Slicing File Template
INFO       [L001] No config_keywords defined for L001
INFO       [L005] No config_keywords defined for L005
INFO       [L006] No config_keywords defined for L006
INFO       [L008] No config_keywords defined for L008
INFO       [L009] No config_keywords defined for L009
INFO       [L015] No config_keywords defined for L015
INFO       [L017] No config_keywords defined for L017
INFO       [L020] No config_keywords defined for L020
INFO       [L021] No config_keywords defined for L021
INFO       [L023] No config_keywords defined for L023
INFO       [L024] No config_keywords defined for L024
INFO       [L025] No config_keywords defined for L025
INFO       [L027] No config_keywords defined for L027
INFO       [L031] No config_keywords defined for L031
INFO       [L032] No config_keywords defined for L032
INFO       [L033] No config_keywords defined for L033
INFO       [L034] No config_keywords defined for L034
INFO       [L035] No config_keywords defined for L035
INFO       [L036] No config_keywords defined for L036
INFO       [L037] No config_keywords defined for L037
INFO       [L039] No config_keywords defined for L039
INFO       [L041] No config_keywords defined for L041
INFO       [L043] No config_keywords defined for L043
INFO       [L044] No config_keywords defined for L044
INFO       [L045] No config_keywords defined for L045
INFO       [L046] No config_keywords defined for L046
INFO       [L048] No config_keywords defined for L048
INFO       [L049] No config_keywords defined for L049
INFO       [L050] No config_keywords defined for L050
INFO       [L051] No config_keywords defined for L051
INFO       [L053] No config_keywords defined for L053
INFO       [L055] No config_keywords defined for L055
INFO       [L056] No config_keywords defined for L056
INFO       [L058] No config_keywords defined for L058
INFO       LEXING RAW (test/fixtures/dialects/spark3/temp.sql)
INFO       Elements to Segments.
INFO       Source-only slices: []
INFO       Lexed tokens: ['SELECT', ' ', '*', ' ', 'FROM', ' ', 'student', ' ', 'WHERE', ' ', 'name', ' ', '=', ' ', '"John Smith"', ';', '\n']
INFO       PARSING (test/fixtures/dialects/spark3/temp.sql)
INFO       
###
#
# Parsed Tree:
#
###
INFO       
[L:  1, P:  1]      |file:
[L:  1, P:  1]      |    base:
[L:  1, P:  1]      |        select_statement:
[L:  1, P:  1]      |            select_clause:
[L:  1, P:  1]      |                keyword:                                      'SELECT'
[L:  1, P:  7]      |                [META] indent:
[L:  1, P:  7]      |                whitespace:                                   ' '
[L:  1, P:  8]      |                select_clause_element:
[L:  1, P:  8]      |                    wildcard_expression:
[L:  1, P:  8]      |                        wildcard_identifier:
[L:  1, P:  8]      |                            star:                             '*'
[L:  1, P:  9]      |            whitespace:                                       ' '
[L:  1, P: 10]      |            [META] dedent:
[L:  1, P: 10]      |            from_clause:
[L:  1, P: 10]      |                keyword:                                      'FROM'
[L:  1, P: 14]      |                whitespace:                                   ' '
[L:  1, P: 15]      |                from_expression:
[L:  1, P: 15]      |                    [META] indent:
[L:  1, P: 15]      |                    from_expression_element:
[L:  1, P: 15]      |                        table_expression:
[L:  1, P: 15]      |                            table_reference:
[L:  1, P: 15]      |                                identifier:                   'student'
[L:  1, P: 22]      |                    [META] dedent:
[L:  1, P: 22]      |            whitespace:                                       ' '
[L:  1, P: 23]      |            where_clause:
[L:  1, P: 23]      |                keyword:                                      'WHERE'
[L:  1, P: 28]      |                [META] indent:
[L:  1, P: 28]      |                whitespace:                                   ' '
[L:  1, P: 29]      |                expression:
[L:  1, P: 29]      |                    column_reference:
[L:  1, P: 29]      |                        identifier:                           'name'
[L:  1, P: 33]      |                [META] dedent:
[L:  1, P: 33]      |                unparsable:                                   !! Expected: 'Nothing...'
[L:  1, P: 33]      |                    whitespace:                               ' '
[L:  1, P: 34]      |                    raw:                                      '='
[L:  1, P: 35]      |                    whitespace:                               ' '
[L:  1, P: 36]      |                    raw:                                      '"John Smith"'
[L:  1, P: 48]      |    statement_terminator:                                     ';'
[L:  1, P: 49]      |    newline:                                                  '\n'

INFO       Found unparsable segment...
INFO       [L:  1, P: 33]      |unparsable:                                                   !! Expected: 'Nothing...'
[L:  1, P: 33]      |    whitespace:                                               ' '
[L:  1, P: 34]      |    raw:                                                      '='
[L:  1, P: 35]      |    whitespace:                                               ' '
[L:  1, P: 36]      |    raw:                                                      '"John Smith"'

INFO       LINTING (test/fixtures/dialects/spark3/temp.sql)
== [test/fixtures/dialects/spark3/temp.sql] LINTING
== [test/fixtures/dialects/spark3/temp.sql] FAIL
L:   1 | P:  33 |  PRS | Line 1, Position 33: Found unparsable section: ' = "John
                       | Smith"'
==== summary ====
files:             1 violations:        1
clean files:       0 unclean files:     1
avg per file:   1.00 unclean rate:   100%
status:         FAIL
All Finished!
```

## Steps to Reproduce
```bash
echo 'SELECT * FROM student WHERE name = "John Smith";' > temp.sql
sqlfluff lint --dialect spark3 temp.sql
```

## Dialect
spark3

## Version
Include the output of `sqlfluff --version` along with your Python version.
If you are using dbt then please additionally include the dbt version.
0.9.1

## Configuration
```
sqlfluff:                0.9.1 python:                 3.10.1
implementation:        cpython dialect:                spark3
verbosity:                   3 templater:               jinja
```

**Comment by jpy-git** [comment]
This makes sense, the reason ANSI parses is for that dialect (and several other dialects) double quotes are for quoted identifiers, which means in your example it's referring to a column called "John Smith" rather than the quoted literal string that you intend.
![image](https://user-images.githubusercontent.com/80432516/149632187-a804411b-fac3-4d52-8b32-278d24949329.png)
In Spark quoted identifiers are quoted using backticks instead, which overrides the ANSI double quotes and therefore stops your example from parsing.

You are correct though that the spark dialect quoted literal regex needs to be updated to include double quotes 👍 Will add this now

**Comment by jpy-git** [comment]
Noticed that spark3 inherits from hive and both these dialects allow single or double quotes. However the way they're currently being handled means that this issue also impacts all quoted literals inherited from ansi that aren't overwritten within these dialects. Will fix this also

**Comment by R7L208** [comment]
Thank you @jpy-git!
